### ISSUE TITLE
Reduce input img size when my custom object is big

### ISSUE THREAD

**Issue by fire717** [ultralytics/yolov5#6063]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi, I'm using awsome yolov5s(v5.0) with my dataset which has only one class and the object is always big (the min size is more than 40pix at size 640, and most object size is range from 100px to 300px).

And in order to reduce the model size and speed up inference, I am trying to using smaller input img size, such as from 640 to 448 or 320.

Because my object is big, so I thought I can get almost the same performance as 640, however I found that the mAP of 448 is lower than 640.

So do you have any suggestions? Such as should I change the model structure to reduce the Receptive field or something like that?

Thx!

**Comment by glenn-jocher** [comment]
@fire717 if you want the best results at 320 you should train directly at 320. Training at 640 but running at 320 will not be quite as accurate, even though it will still work.

**Comment by fire717** [comment]
> @fire717 if you want the best results at 320 you should train directly at 320. Training at 640 but running at 320 will not be quite as accurate, even though it will still work.

Yes! Of course I train at 320.Let me make the question more clear:
My custom dataset only have one class and is big enough at 320 size.
When I train at 640 and test at 640, I can get AP around 0.97,
But if I train at 320 and test at 320, I can get only 0.95 AP.

I thought at 320 it should get 0.97,too. Because my object is easy and big.

So, if the drop from 0.97 to 0.95 is normal, why ? ( Because my object is easy and big)
if not, what should I do to get better performance?

**Comment by glenn-jocher** [comment]
@fire717 👋 Hello! Thanks for asking about improving YOLOv5 🚀 training results. 

Most of the time good results can be obtained with no changes to the models or training settings, **provided your dataset is sufficiently large and well labelled**. If at first you don't get good results, there are steps you might be able to take to improve, but we always recommend users **first train with all default settings** before considering any changes. This helps establish a performance baseline and spot areas for improvement.

If you have questions about your training results **we recommend you provide the maximum amount of information possible** if you expect a helpful response, including results plots (train losses, val losses, P, R, mAP), PR curve, confusion matrix, training mosaics, test results and dataset statistics images such as labels.png. All of these are located in your `project/name` directory, typically `yolov5/runs/train/exp`.

We've put together a full guide for users looking to get the best results on their YOLOv5 trainings below.

## Dataset

- **Images per class.** ≥ 1500 images per class recommended
- **Instances per class.** ≥ 10000 instances (labeled objects) per class recommended
- **Image variety.** Must be representative of deployed environment. For real-world use cases we recommend images from different times of day, different seasons, different weather, different lighting, different angles, different sources (scraped online, collected locally, different cameras) etc.
- **Label consistency.** All instances of all classes in all images must be labelled. Partial labelling will not work.
- **Label accuracy.** Labels must closely enclose each object. No space should exist between an object and it's bounding box. No objects should be missing a label.
- **Background images.** Background images are images with no objects that are added to a dataset to reduce False Positives (FP). We recommend about 0-10% background images to help reduce FPs (COCO has 1000 background images for reference, 1% of the total). No labels are required for background images.

<a href="https://arxiv.org/abs/1405.0312"><img width="800" src="https://user-images.githubusercontent.com/26833433/109398377-82b0ac00-78f1-11eb-9c76-cc7820669d0d.png" alt="COCO Analysis"></a>


## Model Selection

Larger models like YOLOv5x and [YOLOv5x6](https://github.com/ultralytics/yolov5/releases/tag/v5.0) will produce better results in nearly all cases, but have more parameters, require more CUDA memory to train, and are slower to run. For **mobile** deployments we recommend YOLOv5s/m, for **cloud** deployments we recommend YOLOv5l/x. See our README [table](https://github.com/ultralytics/yolov5#pretrained-checkpoints) for a full comparison of all models. 

<p align="center"><img width="700" alt="YOLOv5 Models" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/model_comparison.png"></p>

- **Start from Pretrained weights.** Recommended for small to medium sized datasets (i.e. VOC, VisDrone, GlobalWheat). Pass the name of the model to the `--weights` argument. Models download automatically from the [latest YOLOv5 release](https://github.com/ultralytics/yolov5/releases).
```bash
python train.py --data custom.yaml --weights yolov5s.pt
                                             yolov5m.pt
                                             yolov5l.pt
                                             yolov5x.pt
                                             custom_pretrained.pt
```
- **Start from Scratch.** Recommended for large datasets (i.e. COCO, Objects365, OIv6). Pass the model architecture yaml you are interested in, along with an empty `--weights ''` argument:
```bash
python train.py --data custom.yaml --weights '' --cfg yolov5s.yaml
                                                      yolov5m.yaml
                                                      yolov5l.yaml
                                                      yolov5x.yaml
```


## Training Settings

Before modifying anything, **first train with default settings to establish a performance baseline**. A full list of train.py settings can be found in the [train.py](https://github.com/ultralytics/yolov5/blob/master/train.py) argparser.

- **Epochs.** Start with 300 epochs. If this overfits early then you can reduce epochs. If overfitting does not occur after 300 epochs, train longer, i.e. 600, 1200 etc epochs.
- **Image size.** COCO trains at native resolution of `--img 640`, though due to the high amount of small objects in the dataset it can benefit from training at higher resolutions such as `--img 1280`. If there are many small objects then custom datasets will benefit from training at native or higher resolution. Best inference results are obtained at the same `--img` as the training was run at, i.e. if you train at `--img 1280` you should also test and detect at `--img 1280`.
- **Batch size.** Use the largest `--batch-size` that your hardware allows for. Small batch sizes produce poor batchnorm statistics and should be avoided.
- **Hyperparameters.** Default hyperparameters are in [hyp.scratch.yaml](https://github.com/ultralytics/yolov5/blob/master/data/hyp.scratch.yaml). We recommend you train with default hyperparameters first before thinking of modifying any. In general, increasing augmentation hyperparameters will reduce and delay overfitting, allowing for longer trainings and higher final mAP. Reduction in loss component gain hyperparameters like `hyp['obj']` will help reduce overfitting in those specific loss components. For an automated method of optimizing these hyperparameters, see our [Hyperparameter Evolution Tutorial](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution).

## Further Reading

If you'd like to know more a good place to start is Karpathy's 'Recipe for Training Neural Networks', which has great ideas for training that apply broadly across all ML domains:
http://karpathy.github.io/2019/04/25/recipe/](https://github.com/ultralytics/yolov5/issues/2844#issuecomment-851338384)

Good luck 🍀 and let us know if you have any other questions!

**Comment by zhiqwang** [comment]
Hi @fire717 

> So, if the drop from 0.97 to 0.95 is normal, why ? ( Because my object is easy and big)

This result is quite interesting, what is the map if you test at 640 scale using 320 scale training?

**Comment by fire717** [comment]
@zhiqwang Indeed I did not test that, but I have a similar test to share if you are interested in: 
train at 448, test at 448, got 0.974
train at 448, test at 640, got 0.975
train at 640, test at 640, got 0.981
(and I forget the result of training at 640, test at448, but I remember its not good because I didn't use it after the test. )

Finally I give up trying to reduce the img size, and now I trim a smaller model than yolov5s to speed up inference.

As to the reason for the test, I guess using small size may affect the ability of the model to extract features, even if for big object.

**Comment by zhiqwang** [comment]
Thanks @fire717 , the results look reasonable.
